### ISSUE TITLE
Please make Always on Top in options

### ISSUE THREAD

**Issue by kent79** [Stellarium/stellarium#3379]
As title. Thank you.

**Comment by alex-w** [comment]
Please more info

**Comment by 10110111** [comment]
> Please more info

What more info do you expect? The feature as requested is quite clear: just look at how e.g. Windows Task Manager behaves with respect to the active window. I'm just not sure how cross-platform support for this is. But at least, Qt can request this from the WM if you `setWindowFlags(Qt::WindowStaysOnTopHint)`.

**Comment by kent79** [comment]
> > Please more info
> 
> What more info do you expect? The feature as requested is quite clear: just look at how e.g. Windows Task Manager behaves with respect to the active window. I'm just not sure how cross-platform support for this is. But at least, Qt can request this from the WM if you `setWindowFlags(Qt::WindowStaysOnTopHint)`.

"Always on Top" feature is for windows. You may download "PingInfoView" to check with "Always on Top" feature in options. Thanks

**Comment by 10110111** [comment]
> "Always on Top" feature is for windows.

It's also supported on Linux/X11.

**Comment by kent79** [comment]
> 

You are right. I try to use frontend open Stellarium, When I quit the program back to frontend, the focus will be lost. "Always on Top" may be fix this issue that I think.

**Comment by gzotti** [comment]
In case "frontend" refers to some software, please add a link. This is a too generic word to find what you mean. 
Yes, it would be nice to have a "pin" icon in the Window border. This should be generally available on many Desktop GUI windows.

**Comment by kent79** [comment]
> In case "frontend" refers to some software, please add a link. This is a too generic word to find what you mean. Yes, it would be nice to have a "pin" icon in the Window border. This should be generally available on many Desktop GUI windows.

I am using "playnite"

**Comment by gzotti** [comment]
> 
> You are right. I try to use frontend open Stellarium, When I go into Stellarium or quit the program back to frontend , the focus will be lost in sometimes. "Always on Top" may be fix this issue that I think.

When you launch Stellarium and don't move focus elsewhere while splash is showing, focus should be on Stellarium. 

When you quit the program back to frontend (actually playnite), the focus will no longer be on Stellarium, sure. Where should the focus be then? Probably playnite, but you must change something in playnite then if it is not. 

There is also a difference between showing the window on top and it actually having keyboard focus. 

So, for me it's unclear whether this option would help in your scenario. "Always on top" may be nice on large screens, however.

**Comment by kent79** [comment]
> > You are right. I try to use frontend open Stellarium, When I go into Stellarium or quit the program back to frontend , the focus will be lost in sometimes. "Always on Top" may be fix this issue that I think.
> 
> When you launch Stellarium and don't move focus elsewhere while splash is showing, focus should be on Stellarium.
> 
> When you quit the program back to frontend (actually playnite), the focus will no longer be on Stellarium, sure. Where should the focus be then? Probably playnite, but you must change something in playnite then if it is not.
> 
> There is also a difference between showing the window on top and it actually having keyboard focus.
> 
> So, for me it's unclear whether this option would help in your scenario. "Always on top" may be nice on large screens, however.

Sorry, I just updated my pervious message. The focus is missing while launching Stellarium too. I have tested number of full screen emulators or softwares like as retroarch, mame or PingInfoView. But no any issue on others. It should be Stellarium focus issue.

**Comment by gzotti** [comment]
I just configured my Stellarium to launch in Windowed mode. Splash appears, window appears, then I press one hotkey. It works. So Stellarium has focus after launch. What do you want us to change?

**Comment by kent79** [comment]
> I just configured my Stellarium to launch in Windowed mode. Splash appears, window appears, then I press one hotkey. It works. So Stellarium has focus after launch. What do you want us to change?

If the issue was happened, Stellarium can't launch to Fullscreen mode directly, the screen b it hold on - Windows task bar was shown. I need to click a mouse and then Stellarium will make a focus again and running as normal.

The root cause seems losting focus, so I sugguest making a "Always on Top" options while in full screen mode

**Comment by gzotti** [comment]
Huh? Launching in Fullscreen mode of course has window on top right after start. Unless you switch around during startup.  Therefore, to test what you may might mean, losing focus if started in windowed mode, I reconfigured that. Stellarium has focus in fullscreen and windowed mode right after start, unless you switch to other tasks while splash screen is showing.
